src/html-media-source.js

```diff
@@ -159,6 +159,19 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       });
     };
 
+    this.onPlayerSeeking_ = (event) => {
+      this.sourceBuffers.forEach((sourceBuffer) => {
+        if (sourceBuffer.transmuxer_) {
+          let range = this.player_.buffered();
+          let current = this.player_.currentTime();
+
+          if (current >= range.end(0) || current <= range.start(0)) {
```

Comment:
Since this won't change between buffers, we should do this check first and only do the `forEach` if it passes. 

You'll also want to check `range.length` here.

I think this comparison will also run into problems if there are gaps in the buffer. (history for context) In the past, contrib-hls had pretty complicated logic surrounding segment loading, and part of that complication was gaps and multiple time ranges after seeks. We've since updated that so contrib-hls appends segments in a sequential order and clears the entire buffer when seeking outside the buffered range. What this gave us was the ability for contrib-hls to have the assumption that there is only ever 1 time range of buffered content, and any gaps within that range are inherit to the source and not because we seeked or skipped a segment. However, `buffered()` would still reflect those gaps. For example buffered could look like `[0 -> 10, 11->20]` after appending the first two segments. Seeking to 13 would not reset anything from contrib-hls point of view, but `range.end(0)` would be `10` so this would cause the captions to be reset.

Specifically, contrib-hls will do its seek reset if [this](https://github.com/videojs/videojs-contrib-hls/blob/master/src/master-playlist-controller.js#L1360) `buffered.length === 0`. You'll notice the [findRange function](https://github.com/videojs/videojs-contrib-hls/blob/master/src/ranges.js#L46) it calls also uses a fudge factor to account for rounding of the time range values, so you should probably do something similar here as well